flycheck.el

```diff
@@ -6182,6 +6183,19 @@ Requires DMD 2.066 or newer.  See URL `http://dlang.org/'."
             (or "Warning" "Deprecation") ": " (message) line-end))
   :modes d-mode)
 
+(flycheck-define-checker elixir-dogma
+  "An Elixir syntax checker using the Dogma analysis tool.
+
+See URL `https://github.com/lpil/dogma/'."
+  :command ("mix" "dogma" "--format=flycheck" source-inplace)
```

Comment:
`--format=flycheck`?! :blush:

`dogma` seems to [support JSON](https://github.com/lpil/dogma/blob/master/docs/reporters.md) output, though, and we generally prefer to parse structured output formats, which tend to be less fragile than plain text.

Would you mind to update this pull request to use an `:error-parser` that parses the JSON output of `dogma`?